### ISSUE TITLE
SPSite: Updating owners does not grant those owners site collection admin permissions

### ISSUE THREAD

**Issue by IlleNilsson** [PowerShell/SharePointDsc#1052]
ISSUE TITLE:
SPContentDatabase.FileStorage

    ISSUE DESCRIPTION (this template):
Configure a ContentDatabase to store its document libraries documents on file.

**Comment by IlleNilsson** [comment]
When you let SharePoint DSC set up a site collection, primary and secondary owner, all works well.

If a farm admin manually changes primary or secondary owner of created site collection it seams fine.

Until the next run of DSC, checking for consistency, DSC realizes it should change primary and secondary owner of the site collection according to MOF files.

When DSC changes primary or secondary owner back to what is specified in the MOF files, the reverted  primary and secondary users loses their privileges. They are owners, but lacks privileges as site collection admins.

If we are having problems understanding each other in writing please fell free to contact me directly at Ilian.nilsson@visity.se. We can set up a Skype meeting and I can show you.

BR // Ilian


________________________________

**Comment by ykuijs** [comment]
Do you mean:
- User1 and User2 are site collection owner during site creation
- User3 and User4 are made site collection owner via Central Admin
- DSC restores User1 and User2 as site collection owner
- After the change User3 and User4 no longer have site collection admin permissions to the sitecol

If this is the case, this is as design behavior of SharePoint. As Site Collection Owner, you are automatically added to the Site Collection Administrators group. When owner permissions are removed for a user, SharePoint therefore automatically also revokes the automatic site collection admin permissions. When the user was granted other permissions, these will remain.

**Comment by IlleNilsson** [comment]
Not quite,


  *   User1 and User2 are site collection owner during site creation by DSC (Primary and Secondary Owner)
  *   User3 and User4 are made site collection owner via Central Admin by Farm Admin User (Primary and Secondary Owner)
  *   DSC restores User1 and User2 as site collection owner (Primary and Secondary Owner)
  *   User1 and User2 are site collection owners but does not have site collection owner privileges (they do not have any rights, links for site collection administrative tasks and permission are just vanished in site settings)
BR // Ilian

**Comment by ykuijs** [comment]
Ok, that is strange. Couple of questions:

- Which version of SharePoint are you using?
- What patch level is SharePoint running on?
- Does the ULS log show any errors?
- What are the results of the following command:
```PowerShell
Set-SPSite -Identity [URL] -OwnerAlias [DOMAIN\USER] -SecondaryOwnerAlias [DOMAIN\USER]
```

**Comment by ykuijs** [comment]
Just tested this on SP2016 with the August 2018 CU (build 16.0.4732) and it worked as expected there.

**Comment by IlleNilsson** [comment]
I’ll get back to you on Monday when I’m at the clients site, but I’m running Windows Server 2016, SharePoint 2016 – latest bits by Windows Update and I’m experiencing the problem.

Thanks a bunch for caring and time.

BR // Ilian

**Comment by ykuijs** [comment]
Windows Update doesn't deploy SharePoint Cumulative Updates, just SharePoint security updates. Please check the build number in Central Admin: "Check product and patch installation status"
https://blog.stefan-gossner.com/2014/08/18/sharepoint-patching-demystified/

**Comment by ykuijs** [comment]
@IlleNilsson Any update?

**Comment by DennisL68** [comment]
SharePoint security updates states that they add feature packs.

Anyhow, if just using the cmdlet
Set-SPSite -Identity 'http://www.example.local' -OwnerAlias 'domain\user' 
it looks like the owner has been changed but it hasn't taken effect.

I need to open the CentralAdmin site and click OK to have the ownership be applied.

**Comment by DennisL68** [comment]
This article states that properties Owner and SecondaryContact is considering to be content.
The account I'm running Set-SPSite has no content permissions (it is a farm admin however) and those two properties shows up empty when querying using Get-SPSite.

Should I perhaps reserve one of the owners to a dedicated site collection account and only run scripts affecting the collection using that dedicated account?

That would mean that only one person would get mail notifications from the site collection :/

**Comment by DennisL68** [comment]
@SteveL-MSFT Any updates in this matter?

**Comment by ykuijs** [comment]
I see what the issue is. Will implement a solution in my next Bugfix PR

**Comment by ykuijs** [comment]
This issue is now fixed in dev and will be released with v3.7
